doc/api/webcrypto.md

```diff
@@ -354,12 +354,18 @@ Provides access to the `SubtleCrypto` API.
 added: v15.0.0
 -->
 
-* `typedArray` {Buffer|TypedArray|DataView|ArrayBuffer}
-* Returns: {Buffer|TypedArray|DataView|ArrayBuffer} Returns `typedArray`.
+<!--lint disable maximum-line-length remark-lint-->
+
+* `typedArray` {Buffer|Int8Array|Uint8Array|Uint8ClampedArray|Int16Array|Uint16Array|Int32Array|Uint32Array|BigInt64Array|BigUint64Array}
+* Returns: {Buffer|Int8Array|Uint8Array|Uint8ClampedArray|Int16Array|Uint16Array|Int32Array|Uint32Array|BigInt64Array|BigUint64Array}
```

Comment:
I'm not sure if this makes it more readable. The Web Crypto spec defines the function for any `ArrayBufferView`, it just later defines the function to throw an error when it is an unsupported type of `ArrayBufferView`.

Reply:
Some background on the specs:
- Maybe on early stages there was some confusion between what we call `ArrayBufferView` and [TypedArray](https://tc39.es/ecma262/#sec-typedarray-objects) now.
  - Perhaps this is exactly why this issue happened with Node.js
  - MDN uses `TypedArray` in their definition
  - The proposed fix literally changes `isArrayBufferView` to `isTypedArray`
- `TypeMismatchError` is [deprecated](https://webidl.spec.whatwg.org/#typemismatcherror) in favor of `TypeError`
  - The reason why it wasn't updated in specs is that everyone already implemented it with `TypeMismatchError`
  - Maybe it should be adjusted as well, but Node.js implementation throws the same error for `!isArrayBufferView`

I'm open to being wrong here, but since JS isn't statically-typed, "accepts only A", "accepts anything but immediately throws if it's not A" and "accepts B but immediately throws if it's not A which is subset of B" are equivalent in this context.

Thus, I disagree on mentioning `ArrayBufferView` for this method.


Totally agreed on the type not being more readable. My alternative proposal was:
> If the union type looks too pedantic and ugly, it may be reduced to just `{Buffer|TypedArray}`.

...with explicit mention later that `Float##Array` are forbidden.
My goal here is to avoid any further misinterpretation. If the reader wonders why it looks weird and differs from specs, they will easily find out why it was necessary.

The alternative patch:
```diff
-* `typedArray` {Buffer|TypedArray|DataView|ArrayBuffer}
-* Returns: {Buffer|TypedArray|DataView|ArrayBuffer} Returns `typedArray`.
+* `typedArray` {Buffer|TypedArray}
+* Returns: {Buffer|TypedArray}

 Generates cryptographically strong random values. The given `typedArray` is
 filled with random values, and a reference to `typedArray` is returned.

+The given `typedArray` must be an integer-based instance of {TypedArray},
+i.e. `Float32Array` and `Float64Array` are not accepted.
+
 An error will be thrown if the given `typedArray` is larger than 65,536 bytes.
```
Does it look good?

Reply:
The spec itself is a bit contradictory on this part. Specifically the spec at one point says the API takes "ArrayBufferView", which is defined by Web IDL to include DataView, then immediately after says that it doesn't include DataView. Restricting it down is fine tho and definitely doesn't need to be semver major.

Reply:
Maybe one day those "integer-based arrays" will become a textbook example of how spec can go wrong.

Anyway, updated type definition with more concise version.

Reply:
Thanks, I think I prefer the more concise version :)